extensions/amp-fit-text/0.1/test-e2e/test-amp-fit-text.js

```diff
@@ -43,10 +43,92 @@ describes.endtoend(
       });
 
       await verifyElementStyles(await selectContentDiv('test4'), {
-        'font-size': '20px',
+        'font-size': '19px',
       });
     });
 
+    describe('user initiated dom changes cause recalculations', function () {
```

Comment:
1. I think we can use only one amp-fit-text element at `150x150`.
2. Let's rename for purpose.
3. Let's refactor for clarity.

```suggestion
    describe('user initiated dom changes cause recalculations', function () {
      let contentDiv;
      let originalFontSize;

      async function getFontSize(controller, element) {
        return parseInt(
          await controller.getElementCssValue(element, "font-size"),
          10
        );
      }

      async function nextFrame(controller) {
        // If this method doesn't work, use the previous timeout
        await controller.evaluate(
          () =>
            new Promise((resolve) => {
              requestAnimationFrame(resolve);
            })
        );
      }

      beforeEach(async () => {
        contentDiv = await electContentDiv("#recalculate-at-150-150");
        originalFontSize = await getFontSize(controller, contentDiv);
      });

      it("when box size increases font size should increase", async () => {
        await controller.click(
          await controller.findElement("#recalculate-resize-200-200")
        );
        await nextFrame();
        const updatedFontSize = await getFontSize(controller, contentDiv);
        await expect(updatedFontSize).to.be.lessThan(originalFontSize);
      });

      it("when box size decreases font size should decrease", async () => {
        await controller.click(
          await controller.findElement("#recalculate-resize-100-100")
        );
        await nextFrame();
        const updatedFontSize = await getFontSize(controller, contentDiv);
        await expect(updatedFontSize).to.be.lessThan(originalFontSize);
      });
    });
```

With a simpler HTML structure: 

```html
<h3>Recalculate Font Size</h3>
<amp-fit-text
  id="recalculate-at-150-150"
  width="150"
  height="150"
  [width]="recalculateWidth"
  [height]="recalculateHeight"
  class="ft"
>
  Test to verify font size is correctly recalculated.
</amp-fit-text>

<button
  id="recalculate-resize-200-200"
  on="tap:AMP.setState({recalculateWidth: 200, recalculateHeight: 200})"
>
  Resize to 200x200
</button>

<button
  id="recalculate-resize-100-100"
  on="tap:AMP.setState({recalculateWidth: 100, recalculateHeight: 100})"
>
  Resize to 100x100
</button>
```